### PR TITLE
refactor webhook code logic

### PR DESCRIPTION
**Pull request by allenhaozi**
Signed-off-by: mahao <allenhaozi@gmail.com>

### Ⅰ. Describe what this PR does

upgrade webhook code
from client-go to controller-runtime

### CODE REVIEW

File: charts/fluid/fluid/templates/webhook/MutatingWebhookConfiguration.yaml [hunk 1]
```diff
@@ -21,5 +21,5 @@ webhooks:
     failurePolicy: Ignore
     namespaceSelector:
       matchLabels:
-        Fluid-Injection: enabled
+        fluid.io/inject-fluid-affinity-strategy: "true"
```
Comment (cheyang) [inline charts/fluid/fluid/templates/webhook/MutatingWebhookConfiguration.yaml:24]: How about `fluid.io/inject-fluid-affinity-strategy` to `fluid.io/enable-scheduling-strategy` ?

File: pkg/utils/common.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package utils
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+func GetEnvByKey(k string) (string, error) {
+	if v, ok := os.LookupEnv(k); ok {
```
Comment (cheyang) [inline pkg/utils/common.go:10]: Could you add test for this method?

Comment (allenhaozi) [reply pkg/utils/common.go:24]: done

**Comment by cheyang** [comment]
@yangyuliufeng could you please take a look at this PR? thanks.

File: charts/fluid/fluid/templates/webhook/MutatingWebhookConfiguration.yaml (hunk 1 above)
Comment (yangyuliufeng) [reply charts/fluid/fluid/templates/webhook/MutatingWebhookConfiguration.yaml:24]: how about fluid.io/enable-scheduling-strategy-optimization？

File: tools/certificate.sh [hunk 3]
```diff
@@ -48,8 +48,15 @@ if [ -z "${SERVICE}" ]; then
...
-[ -z "${CERT_DIR}" ] && CERT_DIR=/opt/certs
+if [ -z "${NAMESPACE}" ]; then
+    echo "'--namespace' must be specified"
+    exit 1
+fi
+
+if [ -z "${CERT_DIR}" ]; then
+    echo "'--certDir' must be specified"
+    exit 1
+fi
```
Comment (yangyuliufeng) [inline tools/certificate.sh:59]: why must be specified without permitting default?

File: pkg/webhook/certificate_builder.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+	ca, err := c.genCA(ns, svcName, common.CertificationGenerateFile, cerPath)
+	if err != nil {
+		return err
+	}
+
+	return c.PatchCABundle(webhookName, ca)
+}
+
+func (c *CertificateBuilder) genCA(ns, svc, certFile, certPath string) ([]byte, error) {
```
Comment (yangyuliufeng) [inline pkg/webhook/certificate_builder.go:62]: please add code annotation for these three functions

**Review by yangyuliufeng — COMMENTED** [review]
There's something I don't understand, why we need to split entry function for webhook

File: charts/fluid/fluid/templates/webhook/MutatingWebhookConfiguration.yaml (hunk 1 above)
Comment (allenhaozi) [reply charts/fluid/fluid/templates/webhook/MutatingWebhookConfiguration.yaml:24]: fluid.io/enable-scheduling-strategy
done

File: tools/certificate.sh (hunk 3 above)
Comment (allenhaozi) [reply tools/certificate.sh:59]: As we discussed offline, it's easier and makes it more explicit

File: pkg/webhook/certificate_builder.go (hunk 4 above)
Comment (allenhaozi) [reply pkg/webhook/certificate_builder.go:65]: done

**Review by cheyang — APPROVED** [review]

**Pull request merged**